web/app/scripts/toddow/toddow-directive.js

```diff
@@ -0,0 +1,164 @@
+                        .append('text')
+                          .text(function(d, i) { return theDays[i]; })
+                          .attr('class', 'label month')
+                          .attr('x', 0)
+                          .attr('y', function(d, i) { return i * cellSize + 40; });
+
+                    svg.select('.day').selectAll('g')
+                        .append('text')
+                            .text(function(d, i) {
+                                if(i === 0) {
+                                    return theHours[i] + 'am';
+                                } else if (i === 12) {
+                                    return theHours[i] + 'pm';
+                                } else if (i === 24) {
+                                    return theHours[i] + 'am';
+                                } else {
+                                    return theHours[i];
+                                }
+                            })
+                            .attr('class', 'label hours')
+                            .attr('x', function(d, i) {
+                                var labelVal = theHours[i];
+                                if(typeof labelVal === 'number') {
+                                    return i * cellSize + 27;
+                                } else {
+                                    return i * cellSize + 22;
+                                }
+                            })
+                            .attr('y', 10);
+                }
+
+                /**
+                 * Update all fields of chart with new information and draw in cells with events
+                 */
+                function updateChart(data) {
+                    tooltip = d3.tip().html(function(d) {
+                      var tooltipText = data[d] ? data[d] : '0';
+                      return 'Event count: ' + tooltipText;
+                    });
+                    svg.call(tooltip);
+                    rect.attr('data-hour', function(d) { return formatHourRange(d); })
+                        .datum(formatHourRange)
+                        .on('mouseover', tooltip.show)
+                        .on('mouseout', tooltip.hide);
+
+                    rect.filter(function(d) { return d in data; })
+                        .attr('fill', function(d) { return color(data[d]); });
+                }
+
+                /**
+                 * Helper function to format datetime strings
+                 */
+                function formatHourRange(time) {
+                    return moment(time).format('e:H') + '-' + (+moment(time).format('H') + 1);
+                }
+
+                /**
+                 * Helper function to gather data together into a format more friendly to D3
+                 */
+                function formatData(events) {
+                    /* jshint camelcase: false */
+                    return d3.nest()
+                        .key(function(d) { return formatHourRange(d.occurred_from); })
```

Comment:
It'd be great if the field name could be another bound attribute that would allow setting the field to use as the date (i.e. `d[datetime_field]` instead of `d.occurred_from`). Then this could be used for arbitrary data structures as long as they contained datetime information at the root level.